### ISSUE TITLE
About training errors.

### ISSUE THREAD

**Issue by junwenxiong** [Vanint/DatasetExpansion#1]
I trained on the cifar dataset, but encountered some problems. In dataset_expansion_stable_diffusion_CLIP_batch_optimization_final, bugs appear at this location in the program.
![image](https://github.com/Vanint/DatasetExpansion/assets/35716657/02d3043c-8a08-4bb7-a3dc-465e435506b4)
The model has problems in calculating the gradient, which hopefully can be solved later.
![image](https://github.com/Vanint/DatasetExpansion/assets/35716657/16c922c1-40f3-4ea2-bdb2-0f113cb27619)

**Comment by YifanZhang07** [comment]
Hello, it seems that the issue you're encountering is due to the gradients of channel_noise not being correctly computed. Currently, I'm not sure where the problem lies, as everything is working quite well on my end. Could you please confirm if the data has been loaded properly and whether channel_noise has been successfully utilized?

**Comment by junwenxiong** [comment]
Yes, I ran the code again and found that the data loaded fine, but I'm not sure if channel_noise was used, because from the errors I encountered, it looks like channel_noise was not exploited successfully.  My code is consistent with the repository code without any changes.
![image](https://github.com/Vanint/DatasetExpansion/assets/35716657/df29c40a-4d05-4e75-8ad2-22558f2583a8)

Also, there are still some minor issues with the codebase, some packages are not being imported correctly, as below:
https://github.com/Vanint/DatasetExpansion/blob/330218a6cec304c7da90cae07b580b53ef68bdcd/GIF_SD/CIFAR/dataset_expansion_stable_diffusion_CLIP_batch_optimization_final.py#L31C1-L31C1
https://github.com/Vanint/DatasetExpansion/blob/330218a6cec304c7da90cae07b580b53ef68bdcd/GIF_SD/CIFAR/dataset_expansion_stable_diffusion_CLIP_batch_optimization_final.py#L34C6-L34C6
https://github.com/Vanint/DatasetExpansion/blob/330218a6cec304c7da90cae07b580b53ef68bdcd/GIF_SD/CIFAR/dataset_expansion_stable_diffusion_CLIP_batch_optimization_final.py#L52C5-L53C5

**Comment by YifanZhang07** [comment]
Hi, the channel_noise component should have been successfully applied in the code:

```
channel_noise = Variable(torch.rand([expanded_number_per_sample_each_epoch,channel_noise_dim, 1,1]),  requires_grad=True).cuda() 
channel_noise_bias =  Variable(torch.zeros([expanded_number_per_sample_each_epoch,channel_noise_dim, 1,1]).data.normal_(0, 1),  requires_grad=True).cuda()
samples_temp = samples_temp * (1 + channel_noise) + channel_noise_bias

x_samples = GIF_model.decode_first_stage(samples_temp)
x_samples = torch.clamp((x_samples + 1.0) / 2.0, min=0.0, max=1.0)        
augmented_inputs1 = core.imresize(x_samples, sizes=(224, 224))
image_augmented_features = model.encode_image(augmented_inputs1).float()  
image_augmented_features = image_augmented_features/image_augmented_features.norm(dim=-1, keepdim=True) 
augmented_output_prob = (100.0 * image_augmented_features @  text_classifier.T).softmax(dim=-1)  
augmented_output_entropy = -(augmented_output_prob * torch.log(augmented_output_prob)).sum(dim=1)

delta_entropy = augmented_output_entropy   - original_output_entropy   
delta_entropy = delta_entropy.mean()    
original_output_index1 =  original_output_index.unsqueeze(1)
augmented_output_prob_top1 = torch.gather(augmented_output_prob,1,original_output_index1).mean() 

mean_image_augmented_features = torch.mean(image_augmented_features, 0, keepdim=True).repeat([expanded_number_per_sample_each_epoch,1])
target_mean_image_augmented_features = F.log_softmax(mean_image_augmented_features) 
divergence =  kl_loss(mean_image_augmented_features, target_mean_image_augmented_features)

Score = delta_entropy + augmented_output_prob_top1 + divergence

(channel_noise_grad, channel_noise_bias_grad) = torch.autograd.grad(Score, [channel_noise, channel_noise_bias]) 

channel_noise.data.add_(0.1 * torch.sign(channel_noise_grad)) 
channel_noise_bias.data.add_(0.1 * torch.sign(channel_noise_bias_grad))
```

**Comment by YifanZhang07** [comment]
As for the issue with libraries not being loaded, as per the instructions in the ReadMe, you should copy all the code folders from the ./meta/ directory to the target CIFAR directory. You can do this using the command:
```
cp -r ./meta/* CIFAR
```
This will ensure that all the mentioned dependencies are properly located in the CIFAR directory, which should resolve the issue with missing libraries.

**Comment by junwenxiong** [comment]
After copying the meta folders into the CIFAR directory, the gradient calculation problem did disappear. Very surprised. Thank you for your detailed answer.
